### PR TITLE
[antd_v3.x.x] Added definitions for Select, SelectOption, SelectOptGroup and AutoComplete.

### PR DESCRIPTION
**Pull request by OvestLabs**
- Links to documentation: https://ant.design/components/select/
- Link to GitHub or NPM: https://github.com/ant-design/ant-design
- Type of contribution: addition

Other notes:

### CODE REVIEW

File: definitions/npm/antd_v3.x.x/flow_v0.25.x-/antd_v3.x.x.js [hunk 1]
```diff
@@ -23,7 +23,37 @@ declare module "antd" {
...
+
+  declare export type AutoCompleteProps<T = SelectValue> = {
+    allowClear?: boolean,
+    autoFocus?: boolean,
+    backfill?: boolean,
+    dataSource?: AutoCompleteDataItem[],
+    defaultActiveFirstOption?: boolean,
+    defaultValue?: T,
+    disabled?: boolean,
+    filterOption?: boolean | (input: string, option: React$Element<mixed>) => boolean,
```
Comment (pascalduez) [inline definitions/npm/antd_v3.x.x/flow_v0.25.x-/antd_v3.x.x.js]: Wondering whether `React$Element<mixed>` is a good idea or not.

**Comment by OvestLabs** [comment]
@pascalduez Added tests and modified mixed element. Let me know if it's ok...

File: definitions/npm/antd_v3.x.x/flow_v0.25.x-/antd_v3.x.x.js [hunk 2]
```diff
@@ -427,20 +457,71 @@ declare module "antd" {
...
+    showSearch?: boolean,
+    size?: 'default' | 'small' | 'large' | string,
+    suffixIcon?: React$Node,
+    removeIcon?: React$Node,
+    clearIcon?: React$Node,
+    menuItemSelectedIcon?: React$Node,
+    tokenSeparators?: string[],
+    value?: T,
+    onBlur?: () => void,
+    onChange?: (value: T, option: React$Node | React$Node[]) => void,
```
Comment (pascalduez) [inline definitions/npm/antd_v3.x.x/flow_v0.25.x-/antd_v3.x.x.js]: `React$Node` is enough, it already allows Arrays.

**Comment by pascalduez** [comment]
@OvestLabs Actually, what I was wondering about we the `mixed` part, not really the `React.Element`.
`React.Node` might be fine too, but more permissive.

**Comment by OvestLabs** [comment]
@pascalduez The `SelectOption` component is not exposed, so couldn't use it as the type. I defaulted to `React$Node` for now.

**Pull request merged**